### PR TITLE
Cleanup pending blacklisted items once they expire

### PR DESCRIPTION
**Pull request by shamsimam**
-- cleanup pending blacklisted items once they expire
-- add router change logging

### CODE REVIEW

File: waiter/src/waiter/state.clj [hunk 1]
```diff
@@ -406,18 +406,37 @@
...
-        (cond-> (update-instance-id->blacklist-expiry-time-fn current-state #(dissoc % instance-id))
-          (contains? instance-id->state instance-id)
-          (update-in [:instance-id->state instance-id] update-status-tag-fn #(disj % :blacklisted))))
+  (let [expiry-time (get instance-id->blacklist-expiry-time instance-id)]
+    (if (utils/time-less-than-or-equal-to expiry-time (t/now))
+      (unblacklist-instance current-state update-instance-id->blacklist-expiry-time-fn update-status-tag-fn
+                            instance-id expiry-time)
+      current-state)))
+
+(defn handle-unblacklist-cleanup
```
Comment (sradack) [inline waiter/src/waiter/state.clj]: Rename to `cleanup-blacklist`.  `handle-*` to me implies it's a handler of a message of some kind.

Comment (shamsimam) [reply waiter/src/waiter/state.clj]: renamed to `handle-unblacklist-cleanup-request` to keep it consistent with the other helper methods that have the form `handle-*-request`

File: waiter/src/waiter/utils.clj [hunk 2]
```diff
@@ -472,3 +472,8 @@
   "Parses the request header to determine if debug mode has been enabled."
   [request]
   (boolean (get-in request [:headers "x-waiter-debug"])))
+
+(defn time-less-than-or-equal-to
```
[the remark this answers is not in the document]
Comment (shamsimam) [reply waiter/src/waiter/utils.clj]: Moved it to `state.clj` and call it `expiry-time-reached?`

**Comment by shamsimam** [comment]
RTM

**Pull request merged**